### ISSUE TITLE
graphql.AssertException: interfaceType can't be null

### ISSUE THREAD

**Issue by Togrias** [ExpediaDotCom/graphql-kotlin#136]
The following schema fails to compile with the following reason: "graphql.AssertException: interfaceType can't be null".

Removing inheriting Node2 from Widget2 stops the error, but I don't understand what's wrong.

```
class Query {
	@GraphQLDescription("Fetches an object given its ID")
	suspend fun node(
		@GraphQLContext context: Context,
		@GraphQLDescription("The ID of an object") id: String
	): Node2? = ...
	
	@GraphQLDescription("That's you")
	suspend fun viewer(
		@GraphQLContext context: Context
	): User2? = ...
	
}

@GraphQLDescription("An object with an ID")
interface Node2 {
	
	@GraphQLID
	@GraphQLDescription("The ID of an object")
	val id: String
}

class User2
private constructor(...)
	: Node2 {
	@GraphQLID
	override val id: String ...
	val username get() = data.username
	
	inner class User_WidgetsConnection
	internal constructor(first: Int, after: String?) {
		inner class User_WidgetsEdge
		internal constructor(
			val node: Widget2?,
			val cursor: String
		)
		suspend fun edges(@GraphQLContext context: Context) = ...
		suspend fun pageInfo(@GraphQLContext context: Context) = ...
		suspend fun totalCount(@GraphQLContext context: Context) = ...
	}
	fun widgetsConnection(first: Int, after: String?): User_WidgetsConnection = User_WidgetsConnection(first, after)
	
}
class Widget2 private constructor(...): Node2 {
	
	@GraphQLID
	@GraphQLDescription("The ID of an object")
	override val id: String = ...

	override suspend fun user(@GraphQLContext context: Context): User? = ... //<--- I think this is causing the error.
}
```

**Comment by Togrias** [comment]
Update:

I think it's a bug. The following pattern would cause this error:

```
interface A { ... }
class B: A { fun getC(): C { ... } }
class C: A { fun getB(): B { ... } }
```

I believe a cyclical B -> C -> B where both B and C are members of A would cause this issue.

`graphql.AssertException: interfaceType can't be null`

This is pretty common when implementing Relay specification compliant schemas, because almost everything must implement a "Node" interface. Has anyone successfully created a Relay-compliant schema?

**Comment by smyrick** [comment]
This is resolved in 0.0.31

```kotlin
interface BaseUnion

data class Bar(val barId: Int, var foo: Foo?): BaseUnion

data class Foo(val fooId: Int, var bar: Bar?): BaseUnion
```

![screen shot 2019-01-07 at 13 09 18 pm](https://user-images.githubusercontent.com/2446877/50793584-8a2e1a80-127d-11e9-8ce1-23c33628f93b.png)

![screen shot 2019-01-07 at 13 09 30 pm](https://user-images.githubusercontent.com/2446877/50793591-8d290b00-127d-11e9-89cb-cf5879d6094a.png)
